### PR TITLE
Add bf16 and f16

### PR DESCRIPTION
**Pull request by segeljakt**
This PR adds two more floating point types using the [`half`](https://crates.io/crates/half) Rust-crate.

### CODE REVIEW

File: arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 use crate::compiler::ast;
 use crate::compiler::ast::from::parse::lexer::Token;
 use crate::compiler::info::diags::{Diagnostic, Error};
-use crate::compiler::info::files::{FileId, Loc, ByteIndex};
+use crate::compiler::info::files::{ByteIndex, FileId, Loc};
```
Comment (frej) [inline arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs:4]: Does this belong to this commit?

**Review by frej — APPROVED** [review]

File: arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs (hunk 1 above)
Comment (segeljakt) [reply arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs]: Not sure how that made it in, but I undid it now :)

**Comment by segeljakt** [comment]
> If my calculations are correct, when this baby uses `f88`... you're gonna see some serious shit.

File: arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs (hunk 1 above)
Comment (frej) [reply arc-script/arc-script-core/src/compiler/ast/from/parse/error.rs]: A second pair of eyes is allways good.

**Pull request merged**